src/main/java/mate/academy/webshop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,171 @@
+package mate.academy.webshop.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.webshop.dao.ShoppingCartDao;
+import mate.academy.webshop.lib.Dao;
+import mate.academy.webshop.model.Product;
+import mate.academy.webshop.model.ShoppingCart;
+import mate.academy.webshop.util.ConnectionUtil;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+
+    @Override
+    public ShoppingCart create(ShoppingCart cart) {
+        String query = "INSERT INTO shopping_carts (user_id) VALUES (?)";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                    PreparedStatement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, cart.getUserId());
+            statement.executeUpdate();
+            cart.setId(statement.getGeneratedKeys().getLong(1));
+            putCartProducts(cart);
+            return cart;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create shopping cart", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * FROM shopping_carts WHERE cart_id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                ShoppingCart cart = getCartFromResultSet(resultSet);
+                cart.setProducts(getCartProducts(id));
+                return Optional.of(cart);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new RuntimeException(String.format("Can't get cart id:%d", id), e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        String query = "SELECT * FROM shopping_carts";
+        List<ShoppingCart> cartList = new ArrayList<>();
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                ShoppingCart cart = getCartFromResultSet(resultSet);
+                cart.setProducts(
```

Comment:
try to inline
```suggestion
                cart.setProducts(getCartProducts(cart.getId()));
```
Плюс це трохи б'є по перформансу. Бо ти відкриваєш купу коннекшинів поки відкритий головний connection

Reply:
Цікаво, не знав! Дякую)